Networking/BLIP/BLIPProtocol.hh

```diff
@@ -11,36 +11,61 @@
 //
 
 #pragma once
+#include "fleece/slice.hh"
 #include <stdint.h>
 
 namespace litecore { namespace blip {
 
     // See "docs/BLIP Protocol.md"
 
+    /// The types of messages in BLIP.
     enum MessageType: uint8_t {
-        kRequestType     = 0,  // A message initiated by a peer
-        kResponseType    = 1,  // A response to a Request
-        kErrorType       = 2,  // A response indicating failure
+        kRequestType     = 0,  ///< A message initiated by a peer
+        kResponseType    = 1,  ///< A response to a Request
+        kErrorType       = 2,  ///< A response indicating failure
         kAckRequestType  = 4,  // Acknowledgement of data received from a Request (internal)
         kAckResponseType = 5,  // Acknowledgement of data received from a Response (internal)
     };
 
-    // Array mapping MessageType to a short mnemonic like "REQ".
+    /// Array mapping MessageType to a short mnemonic like "REQ", for logging purposes.
     extern const char* const kMessageTypeNames[8];
 
-
+    /// The flags at the start of a message frame, including the 3 bits containing the type.
     enum FrameFlags: uint8_t {
-        kTypeMask   = 0x07,     // These 3 bits hold a MessageType
-        kCompressed = 0x08,     // Message payload is gzip-deflated
-        kUrgent     = 0x10,     // Message is given priority delivery
-        kNoReply    = 0x20,     // Request only: no response desired
+        kTypeMask   = 0x07,     ///< These 3 bits hold a MessageType
+        kCompressed = 0x08,     ///< Message payload is gzip-deflated
+        kUrgent     = 0x10,     ///< Message is given priority delivery
+        kNoReply    = 0x20,     ///< Request only: no response desired
         kMoreComing = 0x40,     // Used only in frames, not in messages
     };
 
 
-    typedef uint64_t MessageNo;
+    using messageno_t = uint64_t;
+
+    /// A message number. Each peer numbers messages it sends sequentially starting at 1.
+    /// Each peer's message numbers are independent.
+    enum class MessageNo : messageno_t { };
```

Comment:
This is something I've seen before but don't understand, what's the reasoning for declaring this an `enum class`?

Reply:
Additionally, what's the motivation in the first place to make `MessageNo` its own class rather than simply using u64?